### PR TITLE
POWER10 arraycopy inlining and activation

### PR DESCRIPTION
**Pull request by zl-wang**
Deleting inlineVSXArrayCopy: this was a piece of work with experimental characteristics
it bloated the codegen to such an extent that it hurt general workload performance 5-10%
range. It was disabled many years ago. I took this opportunity to delete it completely.
This is duplicated for CS policy in OpenJ9 side.

Adding the POWER10 inlining of arraycopy: this version is applicable to all GC policies
other than CS.  We are using the minimum set of registers as we can.

Adding the capability to call the POWER10 arraycopy helpers too.

Deleting LEArrayCopy related stuffs.

Signed-off-by: Julian Wang <zlwang@ca.ibm.com>

### CODE REVIEW

**Comment by zl-wang** [comment]
@gita-omr  @aviansie-ben please review

File: compiler/p/runtime/PPCArrayCopy.inc [hunk 1]
```diff
@@ -67,10 +67,6 @@
    .globl    __postP10ForwardCopy{DS}
    .globl    .__postP10GenericCopy
    .globl    __postP10GenericCopy{DS}
-   .globl    .__forwardLEArrayCopy
```
[the remark this answers is not in the document]
Comment (zl-wang) [reply compiler/p/runtime/PPCArrayCopy.inc:70]: These are all closely related though ...
I will remove OpenJ9 mentioning, as long as nobody is confused over it.
Fixed the copyright header.

**Comment by fjeremic** [comment]
I'd tend to agree with @aviansie-ben here. Looking at the contents of the change and the message it really does seem like these are 4 semi-related pieces of work going into one commit:

1. Deleting inlineVSXArrayCopy: this was a piece of work with experimental characteristics
it bloated the codegen to such an extent that it hurt general workload performance 5-10%
range. It was disabled many years ago. I took this opportunity to delete it completely.
This is duplicated for CS policy in OpenJ9 side.
2. Adding the POWER10 inlining of arraycopy: this version is applicable to all GC policies
other than CS. We are using the minimum set of registers as we can.
3. Adding the capability to call the POWER10 arraycopy helpers too.
4. Deleting LEArrayCopy related stuffs.

For purposes of reviewing and documenting changes for code archeology when someone views the history in the future I too would like to see these 4 logical pieces of work split into 4 commits and documented as such. For example for the last commit you can explain why we are deleting LEArrayCopy.

**Comment by gita-omr** [comment]
I really think we need to focus more on reviewing the actual correctness of the code. The points above can be considered as recommendations for future PRs and should not hold this work.

**Comment by zl-wang** [comment]
On saturday, I committed them as Filip asked for.

**Comment by fjeremic** [comment]
Thanks for making these changes @zl-wang. It is much appreciated as it will make hte project history much more useful.

**Comment by fjeremic** [comment]
@genie-omr build plinux,aix

**Pull request merged**